currency-converter/tsconfig.node.json

```diff
@@ -0,0 +1,24 @@
+{
+  "compilerOptions": {
+    "tsBuildInfoFile": "./node_modules/.tmp/tsconfig.node.tsbuildinfo",
+    "target": "ES2022",
+    "lib": ["ES2023"],
+    "module": "ESNext",
+    "skipLibCheck": true,
+
+    /* Bundler mode */
+    "moduleResolution": "bundler",
+    "allowImportingTsExtensions": true,
+    "isolatedModules": true,
+    "moduleDetection": "force",
+    "noEmit": true,
+
+    /* Linting */
+    "strict": true,
+    "noUnusedLocals": true,
+    "noUnusedParameters": true,
+    "noFallthroughCasesInSwitch": true,
+    "noUncheckedSideEffectImports": true
+  },
+  "include": ["vite.config.ts"]
+}
```

Comment:
do you know what's these are doing? maybe do a quick chatgpt and explain this to me?

Reply:
It's defining how typescript will use compiler the first part specifies location to store build information, sets JS version, includes built-in libraries, tells how modules should be generated.
Bundler mode specifies compatibility with bundlers like vite
Linting enables strict type checking, ensures no unused variable and parameter is used.